File: 134-gas-station/gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        temp = []
        res = 0
        # for i in range(len(cost)):
        #     res+=(gas[i] - cost[i])
        #     temp.append(gas[i] - cost[i])
        # print(temp)
        if sum(gas) - sum(cost) < 0:
            return -1
        curr_sum = 0
        res = 0
        for i in range(len(gas)):
            curr_sum += (gas[i] - cost[i])
            if curr_sum < 0:
                curr_sum = 0
                res = i + 1
        return res

                    
                

                    
        

File: 134-gas-station/gas-station.java
class Solution {
    public int canCompleteCircuit(int[] A, int[] B) {
        int [] cumDiff = new int[A.length];
        int ans = -1;
        int sum = 0;
        int net = 0;
        for(int i = 0;i<A.length;i++){
            cumDiff[i] = A[i] - B[i];
            sum+=cumDiff[i];
            if(sum >= 0){
                if(ans == -1){
                    ans = i;
                }
            }
            else{
                sum = 0;
                ans = -1;
            }
            net+=cumDiff[i];
            
        }
        if(net >= 0){
            return ans;
        }
        return -1;
    }
}